### PR TITLE
refactor InstitutionsHttpApiSpec

### LINKED ISSUE #584: Refactor InstitutionsHttpApiSpec
Break up big test file into smaller test files to reflect the structure as changed by https://github.com/cfpb/hmda-platform/pull/583

### PR DESCRIPTION
**Pull request by kgudel**
Closes #584

### CODE REVIEW

File: api/src/test/scala/hmda/api/http/InstitutionSpec.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  override def afterAll(): Unit = {
+    system.terminate()
+    val config = ConfigFactory.load()
+    val snapshotStore = new File(config.getString("akka.persistence.snapshot-store.local.dir"))
+    FileUtils.deleteRecursively(snapshotStore)
+  }
+
+  def multiPartFile(contents: String, fileName: String) =
```
Comment (jmarin) [inline api/src/test/scala/hmda/api/http/InstitutionSpec.scala]: Shouldn't this function be included in the `UploadPathSpec` class instead?

Comment (kgudel) [inline api/src/test/scala/hmda/api/http/InstitutionSpec.scala]: This function is used both in `UploadPathSpec` but also in `InstitutionHttpApiSpec`, that's why I put it in `InstitutionSpec`.

File: api/src/test/scala/hmda/api/http/institutions/SubmissionPathsSpec.scala [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import hmda.api.http.InstitutionSpec
+import hmda.api.model.{ErrorResponse, SubmissionStatusWrapper, SubmissionWrapper}
+import hmda.model.fi._
+import hmda.persistence.demo.DemoData
+
+import scala.concurrent.duration._
+
+class SubmissionPathsSpec extends InstitutionSpec {
+
+  override val log: LoggingAdapter = NoLogging
```
Comment (schbetsy) [inline api/src/test/scala/hmda/api/http/institutions/SubmissionPathsSpec.scala]: Could it work to put these lines (initializing `log`, `timeout`, and `ec`) in the parent spec? If so, that would be nice, to reduce duplication.

File: api/src/test/scala/hmda/api/http/InstitutionsHttpApiSpec.scala [hunk 3]
```diff
@@ -1,196 +1,20 @@
...
-import hmda.persistence.institutions.SubmissionPersistence.UpdateSubmissionStatus
-import hmda.persistence.institutions.{ InstitutionPersistence, SubmissionPersistence }
-import org.iq80.leveldb.util.FileUtils
-import org.scalatest.{ BeforeAndAfterAll, MustMatchers, WordSpec }
 
 import scala.concurrent.duration._
 
-class InstitutionsHttpApiSpec extends WordSpec with MustMatchers with ScalatestRouteTest
-    with InstitutionsHttpApi with BeforeAndAfterAll with RequestHeaderUtils {
+class InstitutionsHttpApiSpec extends InstitutionSpec {
```
Comment (schbetsy) [inline api/src/test/scala/hmda/api/http/InstitutionsHttpApiSpec.scala]: It looks like all that's left in this file is authorization-related specs. If that's true, could we rename the spec to be more specific, like `InstitutionsAuthSpec` or similar?

File: api/src/test/scala/hmda/api/http/InstitutionSpec.scala [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import com.typesafe.config.ConfigFactory
+import hmda.api.RequestHeaderUtils
+import hmda.persistence.HmdaSupervisor
+import hmda.persistence.HmdaSupervisor.FindActorByName
+import hmda.persistence.demo.DemoData
+import hmda.persistence.institutions.InstitutionPersistence
+import org.iq80.leveldb.util.FileUtils
+import org.scalatest.{ BeforeAndAfterAll, MustMatchers, WordSpec }
+
+trait InstitutionSpec extends WordSpec with MustMatchers with BeforeAndAfterAll
```
Comment (schbetsy) [inline api/src/test/scala/hmda/api/http/InstitutionSpec.scala]: A minor detail: I think `InstitutionApiSpec` (or something that includes `http` and/or `api` in it) would be a more descriptive name for this spec. There are other specs that cover other institution functionality, like persistence.

Comment (kgudel) [inline api/src/test/scala/hmda/api/http/InstitutionSpec.scala]: Thanks for the suggestion here and with renaming `InstitutionsHttpApiSpec` to something like `InstitutionsAuthSpec`, I was having trouble figuring out the naming and I think that this is right on!

File: api/src/test/scala/hmda/api/http/InstitutionSpec.scala (hunk 1 above)
Comment (jmarin) [inline api/src/test/scala/hmda/api/http/InstitutionSpec.scala]: That makes sense

**Pull request merged**